Updating plugin to 5th version [Step #1]

Sealwing opened the pull request:
# Plugin v5

### Step #1

1. Basic support of ADCM+PostgreSQL added
2. Instruments that manages startup/teardown of ADCM and its environment reworked, some features won't work for a while, some features won't work with new version of ADCM
3. First steps of structure rework
4. ADCM upgrades reworked
5. SSL generation and propagation to ADCM implemented for new flow
6. `--staticimage` is removed and now can be replaced with pairing `--nopull` and `--adcm-image`
7. `--no-postgres` option ADCM added to disable PostgreSQL-oriented setup
8. Fixtures tree reworked
9. ADCMVersionParam is added to wrap requirements for ADCM version (image, tag and with/without postgres launch)
10. `dumpcluster` method as an alternative to archiving PostgreSQL data dir
11. `--launch-option` added to configure ADCM launch/startup/teardown

Sealwing committed:
Plugin v5
Step #1
1. Basic support of ADCM+PostgreSQL added
2. Instruments that manages startup/teardown of ADCM and its environment reworked, some features won't work for a while, some features won't work with new version of ADCM
3. First steps of structure rework

Sealwing committed:
Plugin v5
Step #1.1
1. Return upgrades for old ADCMs
2. Fix SSL generation and propagation to ADCM

Sealwing committed:
Plugin v5
Step #1.2
1. Option to enable no-postgres ADCM added
2. Fixtures tree reworked
3. `--staticimage` is "replaced" with `--nopull` and `--adcm-image`
4. `dumpcluster` as an alternative to archiving postgres data dir
5. ADCMVersionParam is added to wrap requirements for ADCM version (image, tag and with/without postgres launch)

Sealwing committed:
Plugin v5
Step #1.3
1. Minor linters fix

Sealwing committed:
Plugin v5
Step #1.3
1. Linters fix

alexterent commented on src/adcm_pytest_plugin/docker/launchers.py:
```diff
@@ -0,0 +1,298 @@
+            for statement in (
+                "CREATE USER adcm WITH ENCRYPTED PASSWORD 'password';",
+                "CREATE DATABASE adcm OWNER adcm;",
+                "ALTER USER adcm CREATEDB;",
```
Delete role CREATEDB, bc adcm-user has enough grants for adcm-db

Sealwing replied:
done

Sealwing committed:
Plugin v5
Step #1.4
1. `--launch-option` argument added to configure ADCM launch/setup/teardown options
2. ADCM update logic reworked, so further tests will run from "pre-upgrade" image. Because of it, it's also allowed to run ADCM from an image passed to `run`

Sealwing committed:
Plugin v5
Step #1.4.2
1. Linters fix
